sdk/tvOS/OoyalaTVSkinSDK/OOTVGestureManager.m

```diff
@@ -94,63 +106,75 @@ - (BOOL)gestureRecognizerShouldBegin:(UIGestureRecognizer *)gestureRecognizer {
+            seekTo -= FF_SEEK_STEP;
+        }
+        
+        if (seekTo < 0) {
+            seekTo = 0;
+        } else if (seekTo > self.controller.player.duration) {
+            seekTo = self.controller.player.duration;
+        }
+        
+        [self.controller.player seek:seekTo];
+        [self.controller showProgressBar];
   }
 
-  [self.controller.player seek:seekTo];
-  [self.controller showProgressBar];
+- (void)togglePlay:(id)sender {
+    if (self.controller.closedCaptionMenuDisplayed){
+        [self.controller removeClosedCaptionsMenu];
+    }
+      if ([self.controller.player isPlaying]) {
+          [self.controller.player pause];
+      } else {
+          [self.controller.player play];
+      }
+      [self.controller showProgressBar];
 }
 
-- (void)togglePlay:(id)sender {
-  if ([self.controller.player isPlaying]) {
-    [self.controller.player pause];
-  } else {
-    [self.controller.player play];
-  }
-  [self.controller showProgressBar];
+- (void)closedCaptionsSelector: (UITapGestureRecognizer *)sender {
+    [self.controller setupClosedCaptionsMenu];
 }
 
 - (void)onSwipe:(id)sender {
-  if (sender == self.panGesture) {
-    CGPoint currentPoint = [self.panGesture translationInView:self.controller.view];
-    CGFloat viewWidth = self.controller.view.frame.size.width;
-    if (viewWidth == 0) {
-      viewWidth = 1920;
-    }
-    CGFloat seekScale = self.controller.player.duration / viewWidth * SWIPE_TO_SEEK_MULTIPLIER;
-    [self.controller showProgressBar];
-
-    switch (self.panGesture.state) {
-      case UIGestureRecognizerStateBegan:
-        lastPanPoint = currentPoint;
-        lastPlayhead = self.controller.player.playheadTime;
-        break;
-      case UIGestureRecognizerStateChanged: {
-        CGFloat distance = currentPoint.x - lastPanPoint.x;
-        if (abs(distance) > SWIPE_TO_SEEK_MIN_THRESHOLD) {
-          lastPanPoint = currentPoint;
-          lastPlayhead += distance * seekScale;
-          [self.controller.player seek:lastPlayhead];
+    if (!self.controller.closedCaptionMenuDisplayed){
+        if (sender == self.panGesture) {
+            CGPoint currentPoint = [self.panGesture translationInView:self.controller.view];
+            CGFloat viewWidth = self.controller.view.frame.size.width;
+            if (viewWidth == 0) {
+                viewWidth = 1920;
+            }
+            CGFloat seekScale = self.controller.player.duration / viewWidth * SWIPE_TO_SEEK_MULTIPLIER;
+            [self.controller showProgressBar];
+            
+            switch (self.panGesture.state) {
+                case UIGestureRecognizerStateBegan:
+                    lastPanPoint = currentPoint;
+                    lastPlayhead = self.controller.player.playheadTime;
+                    break;
+                case UIGestureRecognizerStateChanged: {
+                    CGFloat distance = currentPoint.x - lastPanPoint.x;
+                    if (abs(distance) > SWIPE_TO_SEEK_MIN_THRESHOLD) {
```

Comment:
use `fabs` instead of `abs` here. I think you didn't touch this code but might as well fix it as you're fixing the rest. Thanks!